network.go

```diff
@@ -1181,7 +1181,8 @@ func (n *network) createEndpoint(name string, options ...EndpointOption) (Endpoi
 	ep.locator = n.getController().clusterHostID()
 	ep.network, err = ep.getNetworkFromStore()
 	if err != nil {
-		return nil, fmt.Errorf("failed to get network during CreateEndpoint: %v", err)
+		logrus.Errorf("failed to get network during CreateEndpoint: %v", err)
+		return nil, ErrNoSuchNetwork(n.Name())
```

Comment:
(left as a comment on the Moby PR as well)

I'd prefer `ep.getNetworkFromStore()` to return this error-type.

While it may be true currently that this is always due to the network not being found, it may not be in future. I don't think we should make the assumption here that this is a `ErrNoSuchNetwork`.

Reply:
While looking for possible other errors it could return, I noticed there some unused error-handling; opened https://github.com/moby/libnetwork/pull/2556

Reply:
I changed return err in `store.go` instead to make it a typed error.